uint64 read_count
uint64 write_count
uint64 read_bytes
uint64 write_bytes
uint64 read_chars 
uint64 write_chars